core/ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/client/BootstrapController.java

```diff
@@ -80,8 +95,22 @@ private void startComponents(Map<String, Provider<Component>> components) {
     private void startWsAgentComponents(EventBus eventBus, final Map<String, Provider<WsAgentComponent>> components) {
         eventBus.addHandler(WorkspaceStartedEvent.TYPE, new WorkspaceStartedHandler() {
             @Override
-            public void onWorkspaceStarted(WorkspaceDto workspace) {
-                startWsAgentComponents(components.values().iterator());
+            public void onWorkspaceStarted(WorkspaceStartedEvent event) {
+                workspaceService.getWorkspace(event.getWorkspace().getId()).then(new Operation<WorkspaceDto>() {
+                    @Override
+                    public void apply(WorkspaceDto ws) throws OperationException {
+                        MachineDto devMachineDto = ws.getRuntime().getDevMachine();
+                        DevMachine devMachine = new DevMachine(devMachineDto);
+                        appContext.setDevMachine(devMachine);
+                        wsAgentStateControllerProvider.get().initialize(devMachine);
+                        startWsAgentComponents(components.values().iterator());
+                    }
+                }).catchError(new Operation<PromiseError>() {
+                    @Override
+                    public void apply(PromiseError err) throws OperationException {
+                        Log.error(getClass(), err.getCause());
```

Comment:
mb it would be better to show some notification here? or call `initializationFailed(...);`